src/app/shared/models/mission.model.ts

```diff
@@ -37,37 +66,50 @@ export class Mission {
     this.endDate = missionEndDate;
     this.unitOfWorkType = missionUnitOfWorkType;
     this.unitOfworkPrice = missionUnitOfWorkPrice;
-    this.secret = secret;
-
-  }
+    this.consultantCompany = consultantCompany || new Company();
+    this.providerCompany = providerCompany || new Company();
+    this.clientCompany = clientCompany || new Company();
+    this.providerBankAccountHolder = providerBankAccountHolder || '';
+    this.providerBankingAgency = providerBankingAgency || '';
+    this.providerBankingDomiciliation = providerBankingDomiciliation || '';
+    this.providerBankIBAN = providerBankIBAN || '';
+    this.providerBankSWIFT = providerBankSWIFT || '';
+    this.consultantBankAccountHolder = consultantBankAccountHolder || '';
+    this.consultantBankingAgency = consultantBankingAgency || '';
+    this.consultantBankingDomiciliation = consultantBankingDomiciliation || '';
+    this.consultantBankIBAN = consultantBankIBAN || '';
+    this.consultantBankSWIFT = consultantBankSWIFT || '';
+    this.consultantFreelance = consultantFreelance;
 
-  static fromFaunaDB(data: ApiFaunaDB) {
-    return new Mission(
-      data.missionCreator,
-      data.consultant,
-      data.consultantEmail,
-      data.clientEmail,
-      data.title,
-      data.clientRef,
-      data.startDate,
-      data.endDate,
-      data.unitOfWorkType,
-      data.unitOfWorkPrice
-    );
   }
 }
-export interface ApiFaunaDB {
-  missionCreator: string;
-  consultant: string;
-  consultantEmail: string;
-  client: string;
-  clientEmail: string;
-  title: string;
-  clientRef: string;
-  startDate: string;
-  endDate: string;
-  unitOfWorkType: string;
-  unitOfWorkPrice: string;
-}
+//   static fromFaunaDB(data: ApiFaunaDB) {
```

Comment:
better to remove instead of comment, if you really need it, you can keep the comment.

Don't forget the github history (if the commit is correctly named ^^)